Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.py

```diff
@@ -0,0 +1,175 @@
+            resp_type='text',
+        )
+        return self.token or ''
+
+    def _call_api(self, method: str, url_suffix: str, json_data: dict = None) -> Dict[str, str]:
+        path = '/'.join([self.api_version, url_suffix])
+        request_string = f'/{path}'
+        return self._http_request(
+            method,
+            url_suffix=path,
+            headers=self._get_headers(request_string),
+            json_data=json_data
+        )
+
+    def get_scopes(self) -> Dict[str, str]:
+        return self._call_api(
+            'GET',
+            url_suffix='scopes'
+        )
+
+    def query_events(self, start_date: str) -> Dict[str, Any]:
+        saas = ['office365_emails']
+        request_data = {
+            'startDate': start_date,
+            'saas': saas
+        }
+        payload = {
+            'requestData': request_data
+        }
+        return self._call_api(
+            'POST',
+            url_suffix='event/query',
+            json_data=payload
+        )
+
+    def get_entity(self, entity: str) -> Dict[str, Any]:
+        return self._call_api(
+            'GET',
+            url_suffix=f'search/entity/{entity}'
+        )
+
+
+def test_module(client: Client):
+    client.get_scopes()
+    demisto.results('ok')
+
+
+def fetch_incidents(client: Client):
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', (datetime.utcnow() - timedelta(hours=1)).isoformat())
+    result = client.query_events(start_date=last_fetch)
+
+    incidents: List[Dict[str, Any]] = []
+    for event in result['responseData']:
+        event_id = event.get('eventId')
+        incidents.append({
+            'name': f'#CP Event: {event_id}',
+            'details': event.get('description'),
+            'occurred': event.get('eventCreated'),
+            'rawJSON': json.dumps(event),
+            'type': 'CheckPointHEC Security Event',
+            'severity': int(event.get('severity')),
+            'dbotMirrorId': event_id,
+            'CustomFields': {
+                'checkpointheccustomer': event.get('customerId'),
+                'checkpointhecsaas': event.get('saas'),
+                'checkpointhecentity': event.get('entityId'),
+                'checkpointhecstate': event.get('state'),
+                'checkpointhectype': event.get('type'),
+            },
+        })
+
+    last = incidents[-1]['occurred'] if incidents else datetime.utcnow().isoformat()
+    demisto.setLastRun({
+        'last_fetch': last
+    })
+    demisto.incidents(incidents)
+
+
+def checkpointhec_get_entity(client: Client, entity: str) -> CommandResults:
+    result = client.get_entity(entity)
+    if row := result['responseData']:
+        return CommandResults(
+            outputs_prefix='CheckPointHEC.Entity',
+            outputs_key_field='entity_id',
+            outputs=row[0]['entityPayload']
+        )
+
+    raise Exception(f'Entity with id {entity} not found')
+
+
+def main() -> None:
```

Comment:
Let's add this to ignore `main` method coverage since it [we're failing coverage](https://circleci.com/gh/demisto/content/708331):

```suggestion
def main() -> None:  # pragma: no cover
```

You can use the [coverage report](https://output.circle-artifacts.com/output/job/50f16fe8-9370-4fb6-b135-ea8fb24d74c9/artifacts/0/artifacts/coverage_report/html/d_4fc29e359444d752_CheckPointHEC_py.html) to see what lines are being hit/missed.